src/object/srv_obj_remote.c

```diff
@@ -150,10 +192,15 @@ ds_obj_remote_update(struct dtx_leader_handle *dlh, void *data, int idx,
 		D_ASSERT(sub->dss_comp == 1);
 		D_ERROR("crt_req_send failed, rc "DF_RC"\n", DP_RC(rc));
 	}
-	return rc;
-
+	sent_rpc = true;
 out:
-	if (rc) {
+	if (shard_iods != NULL && shard_iods != shard_iods_local)
+		D_FREE(shard_iods);
+	if (io_descs != NULL && io_descs != io_descs_local)
+		D_FREE(io_descs);
+	if (offs != NULL && offs != offs_local)
+		D_FREE(offs);
```

Comment:
There is risk: the RPC may be just put into the send queue, but not really sent out yet. Under such case, related RPC body has not been packed before you releasing these data space.

Reply:
hmm, crt_req_send() seems will be sure to put the data on wire.

Reply:
crt_req_send() => crt_context_req_track(), that may return CRT_REQ_TRACK_IN_WAITQ.
@liuxuezhao , would you please to help check CaRT related logic? Thanks!

Reply:
right, as discussed with Di, the data in stack cannot directly pass its pointer to RPC input struct for req_send(). as it possibly return before pack/copy that data so the data in stack may be invalid when cart/HG pack it later.